packages/circuit-ui/hooks/useEscapeKey/useEscapeKey.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright 2021, SumUp Ltd.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useEffect } from 'react';
+
+import { isEscape } from '../../util/key-codes';
+
+/**
+ * Calls a function when the escape key is pressed.
+ * Deactivate it when not used to improve performance.
+ */
+export function useEscapeKey(
+  callback: (event: KeyboardEvent) => void,
+  active = true,
```

Comment:
Is this `active` a common pattern with hooks? I haven't seen it around. Should we document it further, maybe add a "real" usage example? (If it's a story we can even see the performance benefits via the storybook-performance-addon)

Reply:
I think usually the popover content would be a separate component that contains the hook, so when the component isn't rendered, the hook isn't active. 

In our case, the popover content is always rendered, but visually hidden, to enable animations. So while this isn't a common pattern, the performance benefit makes sense to me 🤷🏻‍♂️

The storybook-performance-addon just checks the render performance though, right? I don't think it's able to measure this.

Reply:
> The storybook-performance-addon just checks the render performance though, right? I don't think it's able to measure this.

Yeah,  I guess it would technically be a bit faster since the hook would be skipped when rendering a closed Popover, for example, but it might not make too much sense to add here